src/error.rs

```diff
@@ -110,8 +110,10 @@ cfg_if! {
             core::str::from_utf8(&buf[..idx]).ok()
         }
     } else if #[cfg(target_os = "wasi")] {
-        fn os_err(errno: i32, _buf: &mut [u8]) -> Option<wasi::Error> {
-            wasi::Error::from_raw_error(errno as _)
+        fn os_err(errno: i32, _buf: &mut [u8]) -> Option<wasi::Errno> {
```

Comment:
Can you just change the 
```rust
if #[cfg(unix)] {
```
above to
```rust
if #[cfg(any(unix, target_os = "wasi"))] {
```

and remove the `wasi` specific section here.